src/main/java/com/rackspace/jenkins_nodepool/NodePoolRequestStateWatcher.java

```diff
@@ -95,23 +103,24 @@ public void process(WatchedEvent event) {
             if (event.getType() == Watcher.Event.EventType.NodeDataChanged) {
 
                 // Let's fetch the data from ZK directly so we can read the state value
-                final RequestState state = getStateFromPath();
-                log.fine("Watch event received event type:" + event.getType() + ". State is:" + state);
+                final NodePoolState state = getStateFromPath();
+                log.fine(format("Watch event received event type: %s. State is: %s", event.getType(), state));
 
                 // Are we in the desired state yet? No reason to continue if the request failed.
-                if (state == desiredState || state == RequestState.failed) {
+                if (state == desiredState || state == NodePoolState.FAILED || state == NodePoolState.ABORTED) {
```

Comment:
So previously we carried on waiting when state was ABORTED, but now we accept that as an endstate so we can retry the request or fail?

Reply:
@hughsaunders let's discuss.  I consider the ABORTED state as a specific type of type of failure to fulfill a given request.  As a result, our logic would treat it as a failed attempt and the error handling logic will work as if it was a failure.

Is this the desired behavior or are we expecting us to handle ABORTED via the timeout exception handling route?

Reply:
I think this way is better, was just checking my understanding of the code. Waiting for timeout seems like a waste of time when we already know that the request has been aborted.